stablehlo/dialect/Base.h

```diff
@@ -51,6 +51,16 @@ bool isCompatibleForHloTypeInference(Type tp1, Type tp2);
 // sparsity.
 bool isCompatibleForHloTypeInference(TypeRange tp1, TypeRange tp2);
 
+// Infer single most specific return type from inputTypes with support for
+// bounds. (Size, bound) of each dimension of the return type will be inferred
+// from corresponding dimensions of every inputType by merging them. If
+// `concatenateDims` is set, then the dimensions in `concatenateDims` will be
+// concatenated into return type instead.
+LogicalResult inferMostSpecificType(
+    Optional<Location> location, ValueTypeRange<ValueRange> inputTypes,
+    SmallVectorImpl<Type> &inferredReturnTypes,
+    const SmallVector<int64_t> &concatenateDims = {});
```

Comment:
we will have at most one concat dims so better to take an optional.

Reply:
Yes changed to from `const SmallVector<int64_t> &concatenateDims = {}` to `int64_t concatenateDim =-1`